mantle/kola/harness.go

```diff
@@ -567,8 +568,15 @@ func runExternalTest(c cluster.TestCluster, mach platform.Machine) error {
 		}
 		plog.Debug("Starting kolet run-test-unit")
 		if previousRebootState != "" {
-			plog.Debugf("Setting AUTOPKGTEST_REBOOT_MARK=%s", previousRebootState)
-			c.MustSSHf(mach, "echo AUTOPKGTEST_REBOOT_MARK=%s | sudo tee /run/kola-runext-env", previousRebootState)
+			// quote around the value for systemd
+			contents := fmt.Sprintf("AUTOPKGTEST_REBOOT_MARK='%s'", previousRebootState)
+			plog.Debugf("Setting %s", contents)
+			// This turned out to be easier than trying to quote it for shell
+			quoted := base64.StdEncoding.EncodeToString([]byte(contents))
+			_, _, err := mach.SSH(fmt.Sprintf("echo %s | base64 -d | sudo tee /run/kola-runext-env", quoted))
```

Comment:
How about something like:

```suggestion
			platform.InstallFile(strings.NewReader(contents), mach, "/run/kola-runext-env")
```

?

Reply:
Definitely better, thanks!